patches/node/fix_comment_out_incompatible_crypto_modules.patch

```diff
@@ -9,42 +9,46 @@ with what's exposed through BoringSSL. I plan to upstream parts of this or
 otherwise introduce shims to reduce friction.
 
 diff --git a/src/node_crypto.cc b/src/node_crypto.cc
-index 91cb94d8dbe9db0adbee5e005649188e1ccbcbf9..2429ebf6035de392c6bac18c0b924b995fb3daeb 100644
+index 91cb94d8dbe9db0adbee5e005649188e1ccbcbf9..c3d12dc4cc18888815ff5e2c30a21974322d1faf 100644
 --- a/src/node_crypto.cc
 +++ b/src/node_crypto.cc
-@@ -5191,6 +5191,7 @@ bool DiffieHellman::Init(int primeLength, int g) {
- 
+@@ -5192,11 +5192,11 @@ bool DiffieHellman::Init(int primeLength, int g) {
  bool DiffieHellman::Init(const char* p, int p_len, int g) {
    dh_.reset(DH_new());
-+#if 0
    if (p_len <= 0) {
-     BNerr(BN_F_BN_GENERATE_PRIME_EX, BN_R_BITS_TOO_SMALL);
+-    BNerr(BN_F_BN_GENERATE_PRIME_EX, BN_R_BITS_TOO_SMALL);
++    OPENSSL_PUT_ERROR(BN, BN_R_BITS_TOO_SMALL);
```

Comment:
Question about the PR description:

> We disabled the contents of one of the Diffie-Hellman constructors owing to BoringSSL compatibilities, but I've determined that the functionality itself is compatible; it's just the error throwers that aren't.

Is this saying that the semantics of the error handlers are incompatible (i.e. errors are fundamentally handled differently), or just that the API is incompatible (i.e. `BNerr()` doesn't exist in boringssl)? @codebytere I'm pretty sure that you meant the latter, but want to confirm.

If so, then :+1: on this change being the correct one. Both BNerr and OPENSSL_PUT_ERROR are convenience macros that resolve to `ERR_put_error(..., __FILE__, __LINE__)` so this is a good way to reduce friction when swapping between openssl and boringssl. (And to the comment at the top of this patch file, this is the change that should be upstreamed.)

Reply:
> BNerr() doesn't exist in boringssl

correct